### ISSUE TITLE
Drag and drop a notebook into the pipeline flow editor is not working

### ISSUE THREAD

**Issue by asarantes** [elyra-ai/elyra#2270]
**Describe the issue**
Drag and drop a notebook into the pipeline editor doesn't create a node as expected.

**To Reproduce**
1. Go to File Browser
2. Click on any notebook and drag and drop into editor
3. On the screen nothing happens
4. If you inspect (browser) you see errors. Terminal (OS) shows a 404 GET

**Screenshots or log output**
If applicable, add screenshots or log output to help explain your problem.
Below is the output from the browser inspect -> console.
<details><summary>Log Output</summary>
<pre>
Sending a GET request to http://127.0.0.1:8889/elyra/metadata/runtime-images
requests.js:159 Sending a GET request to http://127.0.0.1:8889/elyra/pipeline/components/local
requests.js:159 Sending a GET request to http://127.0.0.1:8889/elyra/pipeline/components/local/notebook/properties
requests.js:159 Sending a GET request to http://127.0.0.1:8889/elyra/pipeline/components/local/python-script/properties
requests.js:159 Sending a GET request to http://127.0.0.1:8889/elyra/pipeline/components/local/r-script/properties
requests.js:159 Sending a GET request to http://127.0.0.1:8889/elyra/contents/properties/-REDACTED-.ipynb
serverconnection.js:208 GET http://127.0.0.1:8889/elyra/contents/properties/-REDACTED-.ipynb?1635984690068 404 (Not Found)
n @ serverconnection.js:208
n @ serverconnection.js:77
(anonymous) @ requests.js:164
(anonymous) @ requests.js:163
(anonymous) @ requests.js:23
r @ requests.js:19
makeServerRequest @ requests.js:154
(anonymous) @ requests.js:53
(anonymous) @ requests.js:23
r @ requests.js:19
makeGetRequest @ requests.js:52
(anonymous) @ parsing.js:42
(anonymous) @ parsing.js:23
r @ parsing.js:19
getEnvVars @ parsing.js:40
(anonymous) @ PipelineEditorWidget.js:301
(anonymous) @ PipelineEditorWidget.js:27
r @ PipelineEditorWidget.js:23
tt @ PipelineEditorWidget.js:298
(anonymous) @ index.js:1
(anonymous) @ index.js:1
n.addNode @ index.js:1
(anonymous) @ index.js:1
value @ common-canvas.js:42
addFile @ index.js:1
(anonymous) @ PipelineEditorWidget.js:712
(anonymous) @ PipelineEditorWidget.js:705
(anonymous) @ PipelineEditorWidget.js:745
(anonymous) @ PipelineEditorWidget.js:27
r @ PipelineEditorWidget.js:23
bt @ PipelineEditorWidget.js:744
(anonymous) @ Dropzone.js:35
l @ index.es6.js:832
e._evtMouseUp @ index.es6.js:260
e.handleEvent @ index.es6.js:191
Show 5 more frames
untitled1.pipeline:1 Uncaught (in promise) Response {requestPath: "elyra/contents/properties/-REDACTED-.ipynb", type: "basic", url: "http://127.0.0.1:8889/elyra/contents/properties/-REDACTED-.ipynb?1635984690068", redirected: false, status: 404, …}
</pre>

</details>

**Expected behavior**
Go to file browser, drag and drop notebook, and notebook node shows in the editor.

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 3.2.0
 - Operating system: macos 11.3.1
 - Installation source: pip3
 - Deployment type:  local installation

**Comment by lresende** [comment]
I actually saw this happening on Adriano's system, but can't reproduce locally either with master or 3.2.0 which is what he is using. Maybe someone could take a quick look at this and see if you can reproduce it.

**Comment by kevin-bates** [comment]
Hi @asarantes - could you please share some of the characteristics about the name of your notebook file that is redacted in the log output above?  Does it contain any spaces or punctuation (i.e., non-alpha characters)?

Also, could you please reproduce the issue with DEBUG logging enabled by adding the `--debug` flag to your `jupyter` command and provide the log output that is written to the _server_ console window (in which the server was started).  The logging that is performed in that handler only occurs when DEBUG is enabled.  

We need to identify the source of the 404.  It may be due to the regex used to locate the endpoint (which is likely the case if no server-side logging occurs) or some issue with the handler locating the file (in which case we should see some debug logging that the endpoint was found and the handler entered).

If the issue is due to the regex in the handler and the filename does contain some non-alpha characters, it would be a good data point to try renaming the file with plain/vanilla characters to see if the issue still reproduces.  (I'm not suggesting that's the solution in any way, just a data point.)

**Comment by asarantes** [comment]
@kevin-bates The -REDACTED- part. My filename contains dashes "-". My path contains structure similar to this. "@dir/dir_dir_dir/@dir-dir/dir-dir/".
I tried the following:
1) Working as expected. When I create a test directory under the Desktop folder called elyra-test and filenames with and without dash
2) Not working. When I create a test directory under the Desktop folder called @elyra-test and filenames with or without dash.

For 2) see the console logging with -debug
<details><summary>Log Output</summary>
<pre>
[W 2021-11-04 16:54:24.849 ServerApp] 404 GET /elyra/contents/properties/Desktop/@elyra-test/note-1.ipynb?1636070064820 (::1) 1.77ms referer=http://localhost:8889/lab/tree/Desktop/%40elyra-test/untitled.pipeline
</pre>
</details>

In summary, it seems "@" is not supported in the path causing drag and drop to not work. Dashes in the directory and filename work I haven't tried underlines "_".

**Comment by kevin-bates** [comment]
Excellent - thank you for your response @asarantes - that was very helpful.  Just curious, this same notebook (in this location) can be opened by Lab?

I think the solution here is to fix the `path_regex` statement in the handlers definitions: https://github.com/elyra-ai/elyra/blob/5eceb1753e45801bfc356a91582d045cb551daf1/elyra/elyra_app.py#L76

For what its worth, jupyter_server (and notebook) use a `path_regex` of: https://github.com/jupyter-server/jupyter_server/blob/3ecb4e0184b7842d62e6db4357c1fbb22138781d/jupyter_server/base/handlers.py#L969

which is used in their contents endpoint: https://github.com/jupyter-server/jupyter_server/blob/3ecb4e0184b7842d62e6db4357c1fbb22138781d/jupyter_server/services/contents/handlers.py#L319

I'm not sure if this will change the endpoint definition, in addition to the `path_regex` value since it seems the endpoint definition of the jupyter endpoint doesn't include the trailing slash (like that is separator is part of the path definition).

**Comment by asarantes** [comment]
@kevin-bates not a problem. Yes, lab opens the notebook. I successfully created a pipeline and ran it by adding notebooks and associating to these files.

**Comment by kevin-bates** [comment]
@asarantes, @lresende - it does seem related to having an `@` in the path (I added it to a directory name, but haven't tried in the with it in the filename).  However, I just noticed that the first drag/drop did nothing, but the SECOND attempt works (very odd).  Are you seeing the same?

I get this in my server log on the first attempt:
```
[W 2021-11-05 08:21:42.054 ServerApp] 404 GET /elyra/contents/properties/@test/Untitled.ipynb?1636125702049 (::1) 1.53ms referer=http://localhost:8888/lab/tree/%40test/untitled1.pipeline
```
